### PR TITLE
Use setuptools>60 for cpython3.12, skip unicode test on PyPy and 3.12+

### PR DESCRIPTION
**Pull request by mattip**
See issue #5149. This skips the test that cannot run on 3.12+ and is problematic on PyPy. I think there should still be a warning somewhere if the APIs are used.

This also unpins setuptools for 3.12 so the tests can start.

### CODE REVIEW

File: Tools/ci-run.sh [hunk 1]
```diff
@@ -1,5 +1,7 @@
 #!/usr/bin/bash
 
+set -x
+
```
Comment (mattip) [inline Tools/ci-run.sh:4]: Echoing the commands makes it more clear what exactly is going on in the CI logs.

File: runtests.py [hunk 2]
```diff
@@ -2722,7 +2722,8 @@ def runtests(options, cmd_args, coverage=None):
             ('limited_api_bugs.txt', options.limited_api),
             ('windows_bugs.txt', sys.platform == 'win32'),
             ('cygwin_bugs.txt', sys.platform == 'cygwin'),
-            ('windows_bugs_39.txt', sys.platform == 'win32' and sys.version_info[:2] == (3, 9))
+            ('windows_bugs_39.txt', sys.platform == 'win32' and sys.version_info[:2] == (3, 9)),
+            ('python_312.txt', sys.version_info >= (3, 12)),
```
Comment (da-woods) [inline runtests.py:2726]: There's `VERSION_DEP_MODULES` (in runtests.py) that we're using to drop things by version, which is probably better than creating a new "bugs" file.

Comment (mattip) [reply runtests.py]: ok, done

**Comment by mattip** [comment]
python3.12 tests are running. The PyTypeObject in 3.12 gained a `tp_watched` field, so there are lots of [warnings about that](https://github.com/cython/cython/actions/runs/3557754242/jobs/5975979647#step:5:403) which makes it hard to see what else is failing.

File: runtests.py [hunk 3]
```diff
@@ -2722,7 +2725,7 @@ def runtests(options, cmd_args, coverage=None):
             ('limited_api_bugs.txt', options.limited_api),
             ('windows_bugs.txt', sys.platform == 'win32'),
             ('cygwin_bugs.txt', sys.platform == 'cygwin'),
-            ('windows_bugs_39.txt', sys.platform == 'win32' and sys.version_info[:2] == (3, 9))
+            ('windows_bugs_39.txt', sys.platform == 'win32' and sys.version_info[:2] == (3, 9)),
```
Comment (scoder) [inline runtests.py:2728]: While generally a good idea, this change is unrelated to this PR now.
```suggestion
            ('windows_bugs_39.txt', sys.platform == 'win32' and sys.version_info[:2] == (3, 9))
```

**Review by scoder — COMMENTED** [review]
Looks good.

**Comment by scoder** [comment]
Thanks
